src/config_tools/config_tools.cc

```diff
@@ -421,4 +408,121 @@ void ValidateConfig(TritonJson::Value &config, const std::vector<IOConfig> &in_c
   }
   ValidateOutputs(outputs, out_configs);
 }
+
+
+bool is_whitespace(char c) {
+  return std::isspace(static_cast<unsigned char>(c));
+}
+
+
+bool is_sep(char c) {
+  return c == ';' || c == ',';
+}
+
+
+void skip_whitespace(std::string_view &text) {
+  size_t i = 0;
+  while (i < text.size() && is_whitespace(text[i])) ++i;
+  text.remove_prefix(i);
+}
+
+
+void skip_line(std::string_view &text) {
+  auto pos = text.find('\n');
+  if (pos == text.npos) {
+    text = std::string_view();
+  } else {
+    text.remove_prefix(pos+1);
+  }
+}
+
+
+void skip_ignored(std::string_view &text) {
+  skip_whitespace(text);
+  while (!text.empty() && text[0] == '#') {
+    skip_line(text); // remove comment
+    skip_whitespace(text);
+  }
+}
+
+
+void skip_string(std::string_view &text) {
+  // in loop, because string can consist of multiple literals
+  while (!text.empty() && text[0] == '\"') {
+    size_t end = 1;
+    while (end < text.size() && text[end] != '\"') {
+      if (text[end] == '\\') ++end; // escaped character
+      ++end;
+    }
+    text.remove_prefix(end + 1);
+    skip_ignored(text);
+  }
+}
+
+
+void skip_complex(std::string_view &text, char bra, char ket) {
+  if (text.empty()) return;
+  size_t open_bracket = 0;
+  do {
+    if (text[0] == '\"') {
+      skip_string(text);
+    } else {
+      if (text[0] == bra) ++open_bracket;
+      else if (text[0] == ket) --open_bracket;
+      text.remove_prefix(1);
+    }
```

Comment:
This doesn't handle proper nesting - should be something like:
```suggestion
    if (text[0] == '\"') {
      skip_string(text);
    } else if (text[0] == '[') {
      skip_complex(text, '[', ']');
    } else if (text[0] == '{') {
      skip_complex(text, '{', '}');
    } else {
      text.remove_prefix(1);
    }
```